src/Generators/Ics.php

```diff
@@ -55,4 +66,107 @@ protected function generateEventUid(Link $link): string
     {
         return md5($link->from->format(\DateTime::ATOM).$link->to->format(\DateTime::ATOM).$link->title.$link->address);
     }
+
+    protected function addVTimezoneComponents(VCalendar $vcalendar, array $timeZones, \DateTime $from, \DateTime $to)
+    {
+        foreach ($timeZones as $timeZone) {
+            /* @var \DateTimeZone $timeZone */
+            if ($timeZone->getName() !== 'UTC') {
+                $vcalendar->add(
+                    $this->generateVTimeZoneComponent(
+                        $vcalendar,
+                        $timeZone,
+                        $from->getTimestamp(),
+                        $to->getTimestamp()
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * Returns a VTIMEZONE component for a Olson timezone identifier
+     * with daylight transitions covering the given date range.
+     *
+     * Kinldy inspired from https://gist.github.com/thomascube/47ff7d530244c669825736b10877a200
+     * and https://stackoverflow.com/a/25971680
+     *
+     * @param VCalendar $vcalendar
+     * @param \DateTimeZone $timeZone Timezone
+     * @param int $from Unix timestamp with first date/time in this timezone
+     * @param int $to Unix timestap with last date/time in this timezone
+     *
+     * @return Component A Sabre\VObject\Component object representing a VTIMEZONE definition
+     */
+    protected function generateVTimeZoneComponent(VCalendar $vcalendar, \DateTimeZone $timeZone, int $from = 0, int $to = 0)
+    {
+        if (! $from) {
+            $from = time();
+        }
+        if (! $to) {
+            $to = $from;
+        }
+
+        // get all transitions for one year back/ahead
+        $year = 86400 * 360;
+        $transitions = $timeZone->getTransitions($from - $year, $to + $year);
```

Comment:
for some  timezones without DST it's not enough. E.g. latest 4 transactions for `Europe/Minsk` tz are:
```json
[
    {"ts":1269734400,"time":"2010-03-28T00:00:00+0000","offset":10800,"isdst":true,"abbr":"EEST"},
    {"ts":1288483200,"time":"2010-10-31T00:00:00+0000","offset":7200,"isdst":false,"abbr":"EET"},
    {"ts":1301184000,"time":"2011-03-27T00:00:00+0000","offset":10800,"isdst":false,"abbr":"+03"},
    {"ts":2147483647,"time":"2038-01-19T03:14:07+0000","offset":10800,"isdst":false,"abbr":"+03"},
]
```
If you want to generate VTIMEZONE for e.g. 2014-03-01 - it will be empty